dingo/utils.py

```diff
@@ -201,3 +201,127 @@ def get_matrices_of_full_dim_polytope(A, b, Aeq, beq):
+            np.fill_diagonal(filtered_corr_matrix, 1)
+            return filtered_corr_matrix
+        else:
+            np.fill_diagonal(filtered_corr_matrix, 1)
+            return filtered_corr_matrix
+    else:
+        
+        # keep only the lower triangle
+        corr_matrix = np.tril(corr_matrix)
+        # replace diagonal values with 0
+        np.fill_diagonal(corr_matrix, 0)
+    
+        # find indices of correlation matrix where correlation occurs
+        corr_indices = np.argwhere((corr_matrix > pearson_cutoff) | (corr_matrix < -pearson_cutoff))
+            
+        # count reactions with positive or negative correlation based on indicator
+        positive = 0
+        negative = 0
+    
+        # compute copula for each set of correlated reactions
+        for i in range(0, corr_indices.shape[0]):
+            
+            index1 = corr_indices[i][0]
+            index2 = corr_indices[i][1]
+        
+            flux1 = steady_states[index1]
+            flux2 = steady_states[index2]
+                
+            copula = compute_copula(flux1, flux2, cells)
+            rows, cols = copula.shape
+        
+            red_mass = 0
+            blue_mass = 0
+            indicator = 0
+                            
+            for row in range(rows):
+                for col in range(cols):
+                    # values in the diagonal
+                    if ((row-col >= -cop_coeff_1*rows) & (row-col <= cop_coeff_1*rows)):        
+                        # values near the top left and bottom right corner
+                        if ((row+col < cop_coeff_2*rows) | (row+col > cop_coeff_3*rows)):
+                            red_mass = red_mass + copula[row][col]
+                    else:
+                        # values near the top right and bottom left corner
+                        if ((row+col >= cop_coeff_2*rows-1) & (row+col <= cop_coeff_3*rows-1)):
+                            blue_mass = blue_mass + copula[row][col]
+
+            indicator = (red_mass+1e-9) / (blue_mass+1e-9)
+                    
+            # classify specific pair of reactions as positive or negative
+            # correlated based on indicator cutoff 
+            if indicator > indicator_cutoff:
+                positive += 1
+            elif indicator < 1/indicator_cutoff:
+                negative += 1
+            # if they do not overcome the cutoff replace their corresponding
+            # value in the correlation matrix with 0
+            else:
+                filtered_corr_matrix[index1, index2] = 0
+                filtered_corr_matrix[index2, index1] = 0
+                
+            print("Completed process of",i+1,"from",corr_indices.shape[0],"copulas")
+
+                
+        print(positive, "out of", i+1, "copulas were positive correlated based on copula indicator")
+        print(negative, "out of", i+1, "copulas were negative correlated based on copula indicator")
+        
```

Comment:
The prints should be requested by the user I think. These numbers could be part of the output as with a label for each copula: "positive" or "negative"

Reply:
may I create and return a symmetrical matrix, with same dimensions as the correlation matrix, where each position is labeled with +1 (for positive) or -1 (for negative) copula?